### PR TITLE
changed Cluster EDM

### PR DESCRIPTION
**Pull request by konradkusiak97**
This change of cluster EDM aims at making the clusterization algorithm more parallelizable. 

The idea for those changes was triggered by my attempt to port the `measurement_creation` to SYCL. This is still a draft though and definitely some feedback would be very much appreciated :smiley:

pinging @krasznaa :wink:

### CODE REVIEW

File: examples/algorithms/cpu/include/traccc/clusterization/clusterization_algorithm.hpp [hunk 1]
```diff
@@ -66,9 +66,10 @@ class clusterization_algorithm
             auto module = cells_per_event.at(i).header;
 
             // The algorithmic code part: start
-            traccc::cluster_collection clusters_per_module = cc->operator()(
+            traccc::host_cluster_container clusters_per_module = cc->operator()(
```
Comment (krasznaa) [inline examples/algorithms/cpu/include/traccc/clusterization/clusterization_algorithm.hpp:69]: You should rename the variable as well.

In the "new EDM" we don't store the clusters separately for the modules. So the variable name should not try to confuse people about it. :wink:

File: examples/run/cpu/ccl_example.cpp [hunk 2]
```diff
@@ -80,7 +80,7 @@ void print_statistics(const traccc::host_cell_container& data) {
 void run_on_event(traccc::component_connection& cc,
                   traccc::host_cell_container& data) {
     for (std::size_t i = 0; i < data.size(); ++i) {
-        traccc::cluster_collection clusters_per_module =
+        traccc::host_cluster_container clusters_per_module =
```
Comment (krasznaa) [inline examples/run/cpu/ccl_example.cpp:83]: The same here. Let's just go with a simple `clusters` variable name.

File: examples/run/openmp/io_dec_par_example.cpp [hunk 3]
```diff
@@ -62,10 +62,11 @@ traccc::demonstrator_result run(traccc::demonstrator_input input_data,
                 traccc::pixel_segmentation{-8.425, -36.025, 0.05, 0.05};
 
             // The algorithmic code part: start
-            traccc::cluster_collection clusters_per_module =
+            traccc::host_cluster_container clusters_per_module =
```
Comment (krasznaa) [inline examples/run/openmp/io_dec_par_example.cpp:65]: And again. :stuck_out_tongue:

File: examples/run/openmp/par_example.cpp [hunk 4]
```diff
@@ -84,9 +84,10 @@ int par_run(const std::string &detector_file, const std::string &cells_dir,
                 traccc::pixel_segmentation{-8.425, -36.025, 0.05, 0.05};
 
             // The algorithmic code part: start
-            traccc::cluster_collection clusters_per_module =
+            traccc::host_cluster_container clusters_per_module =
```
Comment (krasznaa) [inline examples/run/openmp/par_example.cpp:87]: And once more.

File: io/include/traccc/io/mapper.hpp [hunk 5]
```diff
@@ -185,18 +185,19 @@ measurement_cell_map generate_measurement_cell_map(
         auto module = cells_per_event.at(i).header;
 
         // The algorithmic code part: start
-        cluster_collection clusters_per_module =
+        host_cluster_container clusters_per_module =
```
Comment (krasznaa) [inline io/include/traccc/io/mapper.hpp:188]: You know the drill by now...

**Review by krasznaa — CHANGES_REQUESTED** [review]
Apart from the superficial variable name changes, I'm completely happy with this myself.

Of course we'll need to discuss about it in today's meeting, and you'll also need to resolve the conflict(s) that it collected since yesterday. But other than that, I think this is the way to go...

File: examples/algorithms/cpu/include/traccc/clusterization/clusterization_algorithm.hpp (hunk 1 above)
Comment (konradkusiak97) [reply examples/algorithms/cpu/include/traccc/clusterization/clusterization_algorithm.hpp]: alright, done.

**Comment by krasznaa** [comment]
Ahh... turns out I was able to do that myself... :smile:

**Pull request merged**